From d75184e00e08e97bc8caba6c9677f8f375a051aa Mon Sep 17 00:00:00 2001
From: Hitoshi Harada <umi.tanuki@gmail.com>
Date: Wed, 20 Feb 2013 00:10:56 -0800
Subject: [PATCH 4/5] Bring back V8_SRCDIR and V8_OUTDIR.

Although I decided to delete these options as I thought it was risky
to potentially point to inconsistent include directory and library
directory, but now I'm convinced that that is up to users.  And leaving
some options may help them build plv8 under some circumstances.

Patch by Tony Arkles.
---
 Makefile |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/Makefile b/Makefile
index f5050f4..2d9c6d6 100644
--- a/Makefile
+++ b/Makefile
@@ -41,7 +41,11 @@ REGRESS = init-extension plv8 inline json startup_pre startup varparam json_conv
 ifndef DISABLE_DIALECT
 REGRESS += dialect
 endif
+
 SHLIB_LINK += -lv8
+ifdef V8_OUTDIR
+SHLIB_LINK += -L$(V8_OUTDIR)
+endif
 
 # v8's remote debugger is optional at the moment, since we don't know
 # how much of the v8 installation is built with debugger enabled.
@@ -51,6 +55,10 @@ endif
 OPTFLAGS = -O2
 CCFLAGS = -Wall $(OPTFLAGS) $(OPT_ENABLE_DEBUGGER_SUPPORT)
 
+ifdef V8_SRCDIR
+override CPPFLAGS += -I$(V8_SRCDIR)/include
+endif
+
 all:
 
 plv8_config.h: plv8_config.h.in Makefile
-- 
1.7.10.4

